luigi/static/visualiser/js/graph.js

```diff
@@ -91,35 +145,46 @@ Graph = (function() {
                 rowSizes[depth]++;
                 $.each(n.deps, function(i, dep) {
                     if (nodeIndex[dep]) {
-                        placeNodes(nodes[nodeIndex[dep]], depth + 1);
+                        var next_node = nodes[nodeIndex[dep]]
+                        var depth = (selfDependencies ? depth + 1 : classDepths[next_node.name])
+                        placeNodes(next_node, depth);
                     }
                 });
             }
         }
         placeNodes(nodes[0], 0);
+        console.log("nodes with rows")
+        console.log(nodes)
+        console.log(rowSizes)
```

Comment:
Do we want to leave these console log statement in? I'm not aware if this is desired to do in js code

Reply:
I must have overlooked those. I had way more log statements in and deleted most of them. They are deleted now. I'm not really a JS dev either so can't tell you if we *should* have console.log statements.